<launch>
  <!-- https://github.com/ros-planning/navigation/issues/206 -->
  <rosparam command="delete" ns="move_base" />
  <!-- Map server -->
  <arg name="map_file" default="$(find derobotee)/maps/willo.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
  <node pkg="derobotee" type="shutdown_gmapping" name="shutdown_gmapping"  output="screen"/>
  <arg name="initial_pose_x" default="0.0"/> <!-- Use 17.0 for willow's map in simulation -->
  <arg name="initial_pose_y" default="0.0"/> <!-- Use 17.0 for willow's map in simulation -->
  <arg name="initial_pose_a" default="0.0"/>
  <include file="$(find derobotee)/launch/includes/amcl.launch.xml">
    <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
  </include>

  <include file="$(find derobotee)/launch/includes/move_base.launch.xml"/>
  <include file="$(find frontier_exploration)/launch/boundary.launch"/>
  <node pkg="sparrow_planner" type="sparrow_action_server" name="sparrow_action_server"  output="screen"/>
  <node pkg="derobotee" type="grinder_control" name="grinder_control"  output="screen"/>
</launch>

